src/main.ts

```diff
@@ -0,0 +1,89 @@
+import "./styles/style.css";
+import { getData } from "./util";
+
+type responseType = { key: string; ascii: string; audio: string }[];
+
+const URL =
+  "https://raw.githubusercontent.com/nikhilsnayak/static-data-files/main/drum-kit-data.json";
+const MAX_COUNT = 125;
+
+const keysContainer = document.querySelector<HTMLDivElement>(".keys")!;
+const audiosContainer = document.querySelector<HTMLDivElement>(".audios")!;
+const keyTemplate = <HTMLTemplateElement>(
+  document.getElementById("key-template")!
+);
+const audioTemplate = <HTMLTemplateElement>(
+  document.getElementById("audio-template")!
+);
+
+function renderElements(data: responseType): void {
+  data.forEach((item) => {
+    const key = keyTemplate.content.cloneNode(true) as HTMLElement;
+    const audio = audioTemplate.content.cloneNode(true) as HTMLElement;
+
+    key.querySelector(".key")!.setAttribute("data-key", item.ascii);
+    key.querySelector("kbd")!.textContent = item.key;
+    key.querySelector(".sound")!.textContent = item.audio;
+    key.querySelector(".count")!.textContent = `0/${MAX_COUNT}`;
+    key
+      .querySelector<HTMLDivElement>(".progress-bar")!
+      .style.setProperty("--progress", "0%");
+
+    key.querySelector(".key")!.setAttribute("data-count", "0");
+
+    audio.querySelector("audio")!.src = `/audio/${item.audio}.wav`;
+    audio.querySelector("audio")!.setAttribute("data-key", item.ascii);
+
+    keysContainer.appendChild(key);
+    audiosContainer.appendChild(audio);
+  });
+}
+
+function configure() {
+  window.addEventListener("keydown", (e) => {
+    const audio = document.querySelector<HTMLAudioElement>(
+      `audio[data-key="${e.keyCode}"]`
+    )!;
+
+    const key = document.querySelector<HTMLDivElement>(
+      `.key[data-key="${e.keyCode}"]`
+    )!;
+
+    if (!audio) return;
+    audio.currentTime = 0;
+    audio.play();
+    key.classList.add("playing");
+    const count = parseInt(key.getAttribute("data-count")!);
```

Comment:
We will have several issues when we also use ui elements to store our data, also known as 'model'.  Here, what we need to do is, come up with a good model layer, which will keep track of all keys, and their hit count. It can be a Map data structure, where you have the key as keyboard key, and the value could be all info associated such as its audio file, the hit count and all. So we will be updating model whenever user interacts, and then sync it with the UI.